server/src/main/java/org/elasticsearch/common/logging/ESJsonLayout.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.common.logging;
+
+import org.apache.logging.log4j.core.LogEvent;
+import org.apache.logging.log4j.core.config.plugins.Plugin;
+import org.apache.logging.log4j.core.config.plugins.PluginAttribute;
+import org.apache.logging.log4j.core.config.plugins.PluginFactory;
+import org.apache.logging.log4j.core.layout.AbstractStringLayout;
+import org.apache.logging.log4j.core.layout.ByteBufferDestination;
+import org.apache.logging.log4j.core.layout.PatternLayout;
+
+import java.nio.charset.Charset;
+import java.util.Map;
+
+@Plugin(name = "ESJsonLayout", category = "Core", elementType = "layout", printObject = true)
+public class ESJsonLayout extends AbstractStringLayout {
+
+    private static final String PATTERN = "{" +
+        "\"type\": \"console\", " +
+        "\"timestamp\": \"%d{ISO8601}\", " +
```

Comment:
Currently, all Elasticsearch logs except the GC log emit timestamps without the timezone. This is valid ISO8601, of course, but it would be nice if the time zone could be included as well. 

I see here that the `ESJsonLayout` will also emit ISO8601 timestamps. Will these include timezone information as well?

Reply:
@ycombinator I think this is worth adding - thanks for suggestion. Will include that in this issue